frappe/tests/test_db_query.py

```diff
@@ -564,6 +564,33 @@ def test_reportview_get_aggregation(self):
 			response["keys"], ["field_label", "field_name", "_aggregate_column", "columns"]
 		)
 
+	def test_fieldname_starting_with_int(self):
+		from frappe.core.doctype.doctype.test_doctype import new_doctype
+
+		dt = new_doctype(
+			"dt_with_int_named_fieldname",
+			fields=[{
+				"label": "1field",
+				"fieldname": "1field",
+				"fieldtype": "Int"
```

Comment:
should i change the fieldtype to data? 😛 - as postgres doesn't allow using `like` operator on int type column
ref: https://github.com/frappe/frappe/issues/16722

Reply:
changed to data fieldtype 🙈